clients/src/main/java/org/apache/kafka/clients/admin/AdminClient.java

```diff
@@ -84,9 +85,25 @@ public void close() {
      *
      * @param duration  The duration to use for the wait time.
      * @param unit      The time unit to use for the wait time.
+     * @deprecated Since 2.2. Use {@link #close(Duration)} or {@link #close()}.
      */
+    @Deprecated
     public abstract void close(long duration, TimeUnit unit);
 
+    /**
+     * Close the AdminClient and release all associated resources.
+     *
+     * The close operation has a grace period during which current operations will be allowed to
+     * complete, specified by the given duration.
+     * New operations will not be accepted during the grace period.  Once the grace period is over,
+     * all operations that have not yet been completed will be aborted with a TimeoutException.
+     *
+     * @param timeout  The time to use for the wait time.
+     */
+    public void close(Duration timeout) {
```

Comment:
I asked @ijuma for a second opinion on this approach. It was my suggestion, but it seemed annoying that extensions are forced to implement the deprecated API. He brought up the good point that we have to be able to break extension compatibility when we add new admin APIs anyway. So maybe I had this wrong and we should add the default method to the other `close`. What do you think?

Reply:
> e brought up the good point that we have to be able to break extension compatibility when we add new admin APIs anyway. So maybe I had this wrong and we should add the default method to the other close. What do you think?

Agreed. We have to break extension compatibility in order to push kafka forward. I will update PR later.